packages/next/next-server/server/next-server.ts

```diff
@@ -339,6 +364,68 @@ export default class Server {
       },
     ]
 
+    if (this.customRoutes) {
+      const { redirects, rewrites } = this.customRoutes
+
+      const getCustomRoute = (
+        r: { source: string; destination: string; statusCode?: number },
+        type: 'redirect' | 'rewrite'
+      ) => ({
+        ...r,
+        type,
+        matcher: getCustomRouteMatcher(r.source),
+      })
+
+      const customRoutes = [
+        ...redirects.map(r => getCustomRoute(r, 'redirect')),
+        ...rewrites.map(r => getCustomRoute(r, 'rewrite')),
+      ]
+
+      routes.push(
+        ...customRoutes.map((r, idx) => {
+          return {
+            match: r.matcher,
+            fn: async (req, res, params, parsedUrl) => {
+              let destinationCompiler = pathToRegexp.compile(r.destination)
+              let newUrl = destinationCompiler(params) // /blog/123
+              let newParams = params // { id: 123 }
+              let statusCode = r.statusCode
+              const followingRoutes = customRoutes.slice(idx)
```

Comment:
Should this be the following?
```suggestion
              const followingRoutes = customRoutes.slice(idx + 1)
```

Or are you intentionally including the **currently matched** route?